src/java/org/apache/cassandra/cql3/Sets.java

```diff
@@ -434,9 +311,7 @@ public void execute(DecoratedKey partitionKey, UpdateParameters params) throws I
                 return;
 
             // This can be either a set or a single element
-            Set<ByteBuffer> toDiscard = value instanceof Sets.Value
-                                      ? ((Sets.Value)value).elements
-                                      : Collections.singleton(value.get());
+            List<ByteBuffer> toDiscard = value.getElements();
```

Comment:
Why do you want to keep it?

Reply:
Why do you want to remove it? :-)

Reply:
It is not needed anymore `getElements` does the same thing. If it is a Sets it returns the elements otherwise it returns a list containing the element.